src/server/meme.js

```diff
@@ -1,94 +1,144 @@
 
-      return true;
+      return {
+        isSuccessful: true,
+        value: newMeme,
+      };
     } catch (e) {
       console.log(e);
-      return 'unexpected error when trying to save meme data';
+      return {
+        isSuccessful: false,
+        value: 'unexpected error when trying to save meme data',
+      };
     }
   },
 
-  addComment: async (comment, user) => {
+  // commentData = { meme_id, text }
+  // user = { user_id, username }
+  // returns { isSuccessful, value }
+  addComment: async (commentData, user) => {
     try {
       const meme = await psql('memes')
-        .where({ meme_id: comment.meme_id });
+        .where({ meme_id: commentData.meme_id });
 
       if (meme.length <= 0) {
-        return 'meme with that id does not exist';
+        return {
+          isSuccessful: false,
+          value: 'meme with that id does not exist',
+        };
       }
 
       const newComment = await psql('comments')
         .insert({
           user_id: user.user_id,
-          meme_id: comment.meme_id,
-          text: comment.text,
-          date_created: new Date(),
-        }).returning('*');
+          meme_id: commentData.meme_id,
+          text: commentData.text,
+        }).returning(['comment_id', 'meme_id', 'user_id', 'text', 'date_created']);
+
+      return {
+        isSuccessful: true,
+        value: newComment,
+      };
+    } catch (e) {
+      console.log(e);
+      return {
+        isSuccessful: false,
+        value: 'unexpected error when trying to add comment',
+      };
+    }
+  },
+
+  getMeme: async (memeId, user) => {
+    try {
+      const meme = await baseMemeQuery(psql, user)
+        .where({ 'memes.meme_id': memeId })
+        .first();
 
       return {
-        status: true,
-        comment: newComment,
+        isSuccessful: true,
+        value: meme,
       };
     } catch (e) {
       console.log(e);
-      return 'unexpected error when trying to add comment';
+      return {
+        isSuccessful: false,
+        value: 'unexpected error when trying to retrieve meme',
+      };
     }
   },
 
-  getMemeComments: async (data) => {
+  getMemeComments: async (memeId, earliestId) => {
     try {
       let comments;
 
       /*
-        for lazy loading, on initial load, it will retrieve the latest comments
-        after each subsequent call, expects the id of the earliest comment received
-         in order retrieve the next set of comments
+        For pagination, on initial load, it will retrieve the latest comments
+        after each subsequent call, expects the id of the earliest comment
+        received in order retrieve the next set of comments.
        */
-      if (data.earliest_id) {
+      if (earliestId) {
         comments = await psql('comments')
-          .where({ meme_id: data.meme_id })
+          .where({ meme_id: memeId })
+          .andWhere('comment_id', '<', earliestId)
```

Comment:
should this be .andWhere('comment_id', '>', earliestId)? The earliest id, if provided, is supposed to be the smallest comment id returned in the query, right (ie. give me comments after the comment with earliest id)

Reply:
@adnanrh I made this same mistake too, but how it is right now is correct. Since the newest id will be the largest number.
For example, ```comment_id = 34``` is newer than ```comment_id = 3```